tool/release/createnotes/CreateNotes.kt

```diff
@@ -21,27 +21,36 @@
 
 package com.vaticle.dependencies.tool.release.createnotes
 
+import java.nio.file.Files
+import java.nio.file.Path
 import java.nio.file.Paths
 import kotlin.io.path.notExists
 
 fun main(args: Array<String>) {
     val bazelWorkspaceDir = Paths.get(getEnv("BUILD_WORKSPACE_DIRECTORY"))
     val githubToken = getEnv("CREATE_NOTES_TOKEN")
-    if (args.size != 5) throw RuntimeException("org, repo, version, commit, and template must be supplied")
-    val (org, repo, version, commit, templateFileLocation) = args
+    if (args.size != 5) throw RuntimeException("org, repo, commit, version, and template must be supplied")
+    val (org, repo, commit, version, templateFileLocation) = args
     val templateFile = bazelWorkspaceDir.resolve(templateFileLocation)
     if (templateFile.notExists()) throw RuntimeException("Template file '$templateFile' does not exist.")
 
-    println("Repository: $org/$repo")
-    println("Commit: $commit")
+    println("Commit: $org/$repo@$commit")
     println("Version: $version")
 
-    val commits = getCommits(org, repo, Version.parse(version), commit, bazelWorkspaceDir, githubToken)
-    println("There are ${commits.size} commits to be collected.")
-    val commitDescriptions = getCommitDescriptions(org, repo, commits, githubToken)
-    writeReleaseNoteMd(commitDescriptions, templateFile)
+    val commits = collectCommits(org, repo, commit, Version.parse(version), bazelWorkspaceDir, githubToken)
+    println("Found ${commits.size} commits to be collected into the release note.")
+    val notes = collectNotes(org, repo, commits, githubToken)
+    writeNotesMd(notes, templateFile)
 }
 
 private fun getEnv(env: String): String {
     return System.getenv(env) ?: throw RuntimeException("'$env' environment variable must be set.")
-}
\ No newline at end of file
+}
+
+private fun writeNotesMd(notes: List<Note>, releaseTemplateFile: Path) {
+    val template = String(Files.readAllBytes(releaseTemplateFile), Charsets.UTF_8)
+    if (!template.matches("[\\s\\S]*${Constant.releaseTemplateRegex.pattern}[\\s\\S]*".toRegex()))
+        throw RuntimeException("The release-template does not contain the '${Constant.releaseTemplateRegex}' placeholder")
+    val markdown = template.replace(Constant.releaseTemplateRegex, Note.toMarkdown(notes))
```

Comment:
@lolski side note: do we need placeholder to be a regex? can we make it a static placeholder string? e.g. `"{ release notes }"` with exactly one space

Reply:
I was borrowing the idea from the old script. I kinda see the point - intuitively when it comes to "closure"  syntax like this one, people would expect the number of whitespace to have no effect.

I'd say let's keep it for now and not redesign the placeholder command in this PR.